### PR TITLE
Custom scripts- demonstration 2022 with torch functions and monkeypatched custom actions "fireball" and "slash"

### LINKED ISSUE #1648: Integrate gold.c elements with demonstration
Include chest and torch functionalities using Lua from the gold.c and dark_room.c files respectively into the current demonstration (pending pull request). This issue will be closed once we have successfully updated the demonstration.

### LINKED ISSUE #1655: Custom scripts- demonstration 2022 with torch functions and monkeypatched custom actions "fireball" and "slash"
We want to update our demonstration to include the torch function that can be interacted with through LUA (user can be lit and then unlit). Additionally, we are monkey patching custom actions called "fireball" and "slash" instead of asking the action management team to hardcode it into Chiventure which may lead to convoluting the actions list in Chiventure.

### PR DESCRIPTION
**Pull request by RamarkoB**
This is a working demo with Conditionals, Multiple WDL files, and the Torch. In short, It comprises of nearly everything custom scripts has accomplished this quarter.  The one thing issue this pull request still has is that the Lua Functions are not compiling correctly. Once Togay takes a look at this to correct the Luad functions to match the Refactoring, it should be good to go!

### CODE REVIEW

**Comment by RamarkoB** [comment]
For some reason, [8ff23d6](https://github.com/uchicago-cs/chiventure/pull/1651/commits/8ff23d62acf46fbd5abb30bc46ffc4c552e531a9) broke the build, but i removed one line and added it back, and suddenly, it all compiled correctly. Github is crazy

**Review by mxu24 — APPROVED** [review]
Past issues all fixed, demonstration functions correctly with all new custom actions, integrated Lua and WDL content.

File: src/custom-scripts/examples/demo_2022.c [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    dc.c = string_num;
+    togay = obj_add_arg(togay,dc,CHAR_TYPE);
+    char* custom_string = (char*)malloc(100);
+    data_t temp = arg_t_get(togay);
+    custom_string = temp.s;
+
+    obj_t *obj_store;
+
+    char wizard[] = "Wizard Class", warrior[] = "Warrior Class";
+    if(strcmp(wizard,custom_string)== 0){
```
Comment (ecoble) [inline src/custom-scripts/examples/demo_2022.c]: Spacing between ( and {

File: src/custom-scripts/examples/loadwdl2.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    char string_num;
+    printf("Enter either 1 or 2 (1 for wizard class, 2 for warrior class): ");
+    scanf("%c", &string_num); 
+    dc.c = string_num;
+    togay = obj_add_arg(togay,dc,CHAR_TYPE);
+    char* custom_string = (char*)malloc(100);
+    data_t temp = arg_t_get(togay);
+    custom_string = temp.s;
+    char wizard[] = "Wizard Class", warrior[] = "Warrior Class";
+    if(strcmp(wizard,custom_string)== 0){
```
Comment (ecoble) [inline src/custom-scripts/examples/loadwdl2.c]: Style spacing

**Review by ecoble — CHANGES_REQUESTED** [review]
Small style changes

**Review by ecoble — APPROVED** [review]

**Comment by RamarkoB** [comment]
We were able to both integrate the dark_room.c demo with the torch, and add custom actions for fireball and slash into a single demo, solving two issues with a single pull request. Moving forward, we want to integrate more small demos into a single, larger demo that we can present

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Great job creating this demo and debugging the issues! Please try to make sure the PR is ready to go before submitting it.

**Pull request merged**